From dde05e26be153002c7e52cc8f410e2e7be123707 Mon Sep 17 00:00:00 2001
From: Philip Chimento <philip.chimento@gmail.com>
Date: Wed, 29 Apr 2015 22:20:47 -0700
Subject: [PATCH] Fix non-libtool linker flags on Darwin

Darwin's linker doesn't like "-rpath=path"; instead pass "-rpath path".
Additionally, there is no equivalent for "--no-as-needed" (it seems to do
the right thing by default?)
---
 giscanner/ccompiler.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/giscanner/ccompiler.py b/giscanner/ccompiler.py
index 1c66aac..25ef55c 100644
--- a/giscanner/ccompiler.py
+++ b/giscanner/ccompiler.py
@@ -69,11 +69,11 @@ class CCompiler(object):
                 else:
                     args.append(str.lower(ns) + '.lib')
             else:
-                args.append('-Wl,-rpath=.')
+                args.append('-Wl,-rpath,.')
 
         # Ensure libraries are always linked as we are going to use ldd to work
         # out their names later
-        if not libtool and pkgconfig_msvc_flags == '':
+        if not libtool and pkgconfig_msvc_flags == '' and not sys.platform == 'darwin':
             args.append('-Wl,--no-as-needed')
 
         for library in libraries:
@@ -96,7 +96,7 @@ class CCompiler(object):
                         args.append('-rpath')
                         args.append(library_path)
                     else:
-                        args.append('-Wl,-rpath=' + library_path)
+                        args.append('-Wl,-rpath,' + library_path)
 
     def get_external_link_flags(self, args, libraries, pkgconfig_msvc_flags):
         # An "external" link is where the library to be introspected
-- 
2.3.2 (Apple Git-55)

